test/test_storage.py

```diff
@@ -218,6 +219,13 @@ class TestSQLiteStorage(StoragePermutations, BaseTestCase):
     def create_storage(self):
         return ops.storage.SQLiteStorage(':memory:')
 
+    def test_permissions(self):
```

Comment:
Can we also add a test for the existing-file case now that we have a separate code path for that?